File: python/euler/bin/p0001.py
# project euler: problem 1


def compute(ulimit: int) -> str:
    def sum_of_multiples(n: int) -> int:
        tmp = (ulimit - 1) // n
        return (1 + tmp) * tmp // 2 * n

    return str(sum_of_multiples(3) + sum_of_multiples(5) - sum_of_multiples(15))


def solve() -> str:
    return compute(1000)

File: typescript/euler/bin/p0001.ts
// project euler: problem 1

export const compute = (limit: number): string => {
  const sumMultiples = (n: number): number => {
    const tmp = Math.trunc((limit - 1) / n);
    return Math.trunc((1 + tmp) * tmp * n / 2);
  };

  return String(sumMultiples(3) + sumMultiples(5) - sumMultiples(15));
};

export const solve = (): string => compute(1_000);
